Files/View Models/SettingsViewModel.cs

```diff
@@ -93,6 +97,61 @@ public DefaultLanguageModel DefaultLanguage
             }
         }
 
+        private SortOption _directorySortOption = SortOption.Name;
+        private SortDirection _directorySortDirection = SortDirection.Ascending;
+        public SortOption DirectorySortOption
+        {
+            get => Get(_directorySortOption);
+            set
+            {
+                Set(ref _directorySortOption, value);
+                localSettings.Values[LocalSettings.SortOption] = (byte)value;
+		App.CurrentInstance?.FilesystemViewModel?.UpdateSortOptionStatus();
+                App.CurrentInstance?.FilesystemViewModel?.OrderFiles();
+            }
+        }
+
+        public SortDirection DirectorySortDirection
+        {
+            get => Get(_directorySortDirection);
+            set
+            {
+                if (value != _directorySortDirection)
+                {
+                    Set(ref _directorySortDirection, value);
+                    localSettings.Values[LocalSettings.SortDirection] =(byte)value;
+		    App.CurrentInstance?.FilesystemViewModel?.UpdateSortDirectionStatus();
+                    App.CurrentInstance?.FilesystemViewModel?.OrderFiles();
+                }
+            }
+        }
+
+        public void DetectSortOption()
+        {
+            if (localSettings.Values[LocalSettings.SortOption] == null)
+            {
+                _directorySortOption = SortOption.Name;
+                localSettings.Values[LocalSettings.SortOption] = SortOption.Name;
+            }
+            else
+            {
+                _directorySortOption = (SortOption)localSettings.Values[LocalSettings.SortOption];
+            }
+        }
```

Comment:
Can you try using a format similar to this for storing the settings?
```
private string _AppDataPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);

public string AppDataPath
{
    get => _AppDataPath;
    set => Set(ref _AppDataPath, value);
}
```

Reply:
This is for detecting the sort option from AppData.

Reply:
This method does not require you to "detect" the setting as it does it for you.

Reply:
LocalSettings doesn't support Enums.

Reply:
Done